### PR TITLE
Installing Elasticsearch on linux machine

### PR DESCRIPTION
**Pull request by daudahabbasi**
Added the images and user guide to install Elasticsearch on linux machine.

### CODE REVIEW

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+     - **gateway.recover_after_nodes**: Set this to *1*.
+
+     - **node.master**: Set this to *true*. For more information on master nodes, see [Configuring the master nodes](xref:Configuring_master_Elasticsearch_nodes).
+
+1. Make the following adjustments in */etc/sysconfig/elasticsearch*:
+
+   - **MAX_LOCKED_MEMORY**: Set this to *unlimited*.
+
+   - **JAVA_HOME**: Set this to *java-11-openjdk-11.0.14.0.9-1.el7_9.x86_64* a custom java path to be used.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:84]: This isn't clear to me. Could you rephrase this?

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+# Installing Elasticsearch on a Linux machine
+
+If you want to use an Elasticsearch cluster for your DMS (which is required to use the [Cassandra Cluster](xref:Migrating_the_general_database_to_a_DMS_Cassandra_cluster) feature), install Elasticsearch on a Linux machine as detailed below.
+
+1. Install the Elasticsearch software on  the Linux machine as described under [Installing from the RPM repository](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/rpm.html#rpm-repo) in the official Elasticsearch documentation.
+
+1. Make sure the firewall ports are open for Elasticsearch. Elasticsearch operates on TCP port 9200 and TCP port 9300.
+
+   - There is a default firewall on RHEL, but this is disabled by default. To enable the firewall, use the following command:
```
Comment (ThibaultHeylen) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:13]: Can we make it more general and not only for RHEL? Other guides that were written were all for Ubuntu server. Might be confusing if we have several distro's in our guidelines.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+---
+uid: Installing_Elasticsearch_on_Linux
+---
+
+# Installing Elasticsearch on a Linux machine
+
+If you want to use an Elasticsearch cluster for your DMS (which is required to use the [Cassandra Cluster](xref:Migrating_the_general_database_to_a_DMS_Cassandra_cluster) feature), install Elasticsearch on a Linux machine as detailed below.
+
+1. Install the Elasticsearch software on  the Linux machine as described under [Installing from the RPM repository](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/rpm.html#rpm-repo) in the official Elasticsearch documentation.
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:9]: We use Ubuntu in the docs normally. Change to Debian for Consistency.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+1. Make sure the firewall ports are open for Elasticsearch. Elasticsearch operates on TCP port 9200 and TCP port 9300.
+
+   - There is a default firewall on Linux, but this is disabled by default. To enable the firewall, use the following command:
+
+     `$ systemctl start firewalld.service`
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:15]: Firewalld is not installed by default on Ubuntu (Debian), invalidating the entire section. 
There might be something else, but this would need to be looked into further.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+1. Configure the *elasticsearch.yml* files.
+
+   - Make sure the Elasticsearch service is stopped. To stop the Elasticsearch service, use the following command:
+
+     `$ sudo systemctl stop elasticsearch.service`
+
+   - To change *elasticsearch.yml* files, you can use the following command:
+
+     `$ vi /etc/elasticsearch/elasticsearch.yml`
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:51]: When using vi, might as well link to a vi reference for dummies: e.g. https://www.cs.colostate.edu/helpdocs/vi.html 
Can also recommend nano as it is more simple to work with. Vi is not a very intuitive editor.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+     - **network.host**: The IP address of the node.
+
+     - **http.port**: Set this to *9200*.
+
+     - **discovery.zen.ping.unicast.hosts**: The IP addresses of all the nodes in your Elasticsearch cluster. Elasticsearch nodes use this list of hosts to find each other and learn the topology of the ring.
+
+       > [!NOTE]
+       >
+       > - We recommend a cluster of three nodes, preferably in different racks.
+       > - To add a node to an existing cluster, see [Adding an Elasticsearch cluster node](xref:Configuring_Elasticsearch_node_add).
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:72]: Is a very specific setup. I don't believe we recommend a cluster of precisely three nodes, we recommend a cluster of minimum 3 nodes. (because of the master node configuration)

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+     - **discovery.zen.minimum_master_nodes**: Set this to *2*. For more information on master nodes, see [Configuring the master nodes](xref:Configuring_master_Elasticsearch_nodes).
+
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:75]: too specific. Refer too master node configuration only.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+     - **gateway.recover_after_nodes**: Set this to *1*.
+
+     - **node.master**: Set this to *true*. For more information on master nodes, see [Configuring the master nodes](xref:Configuring_master_Elasticsearch_nodes).
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:78]: too specific. Refer too master node configuration only.

File: user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+   - **MAX_LOCKED_MEMORY**: Set this to *unlimited*.
+
+   - **JAVA_HOME**: Set this to the custom Java path to be used for Elasticsearch, e.g. *java-11-openjdk-11.0.14.0.9-1.el7_9.x86_64*.
+
+1. Make the following adjustment in */usr/lib/systemd/system/elasticsearch.service*:
+
+   - **LimitMEMLOCK**: Set this to *infinity*. This must be set under the *Service* tag.
+
+1. Set the maximum Java Heap Size under */etc/elasticsearch/jvm.options*. For more information, see [Setting the heap size](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/heap-size.html).
+
```
Comment (SLC-Laurens-Vercruysse) [inline user-guide/Advanced_Functionality/Databases/Elasticsearch_database/Installing_Elasticsearch_on_Linux.md:91]: maybe add an !Important note here that Elasticsearch uses quite a bit of memory and that at least 8 Gb of Heap size should be specified, and more may be required depending on the demands of the system.

**Comment by MariekeGO** [comment]
@SLC-Laurens-Vercruysse, I've adjusted this PR so that it now fits in the overhauled structure of the database section. Could you confirm whether it is now OK to merge?

**Review by SLC-Laurens-Vercruysse — APPROVED** [review]

**Review by MariekeGO — APPROVED** [review]
